ega-data-api-commons/src/main/java/eu/elixir/ega/ebi/commons/config/MyAccessTokenConverter.java

```diff
@@ -174,12 +220,60 @@ public OAuth2Authentication extractAuthentication(Map<String, ?> map) {
         return new OAuth2Authentication(request, user);
     }
 
+    private String getEGAAccountId(final Map<String, ?> map) {
+        return userDetailsService
+                .getEGAAccountId((String) map.get("user_id"))
+                .orElseGet(() -> userDetailsService
+                        .getEGAAccountIdForElixirId((String) map.get("sub"))
+                        .orElse(""));
+    }
```

Comment:
Not a fan of the last initialization with empty String, as fas as I could see the ga4gh service is not prepared for that eventuality either. It may be for the best to control it and decide what to do if we could not get an EGA account id

Reply:
Done.